coffeecard/CoffeeCard.Library/Services/v2/PurchaseService.cs

```diff
@@ -28,12 +28,21 @@ public PurchaseService(CoffeeCardContext context, IMobilePayPaymentsService mobi
 
         public async Task<InitiatePurchaseResponse> InitiatePurchase(InitiatePurchaseRequest initiateRequest, User user)
         {
-            var product = await _context.Products.FindAsync(initiateRequest.ProductId);
+            var product = await _context.Products.Include(p => p.ProductUserGroup).FirstAsync(p => p.Id == initiateRequest.ProductId);
```

Comment:
`FirstAsync` will throw an exception if no entity is found. We should use `FirstOrDefaultAsync` as this will just return `null` if no entity is found by the ProductId